examples/tfidf.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright 2016 Husky Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <string>
+#include <utility>
+#include <vector>
+
+
+#include "boost/utility/string_ref.hpp"
+#include "boost/tokenizer.hpp"
+#include "mongo/bson/bson.h"
+#include "mongo/client/dbclient.h"
+
+#include "base/serialization.hpp"
+#include "core/engine.hpp"
+#include "io/input/mongodb_inputformat.hpp"
+#include "lib/aggregator_factory.hpp"
+
+class Document {
+  public:
+    using KeyT = std::string;
+    Document() = default;
+    explicit Document(const KeyT& t) : title(t){}
+    KeyT title;
+    std::vector<double> tf;
+    std::vector<double> tf_idf;
+    std::vector<std::string> words;
+    int total_words = 0;
+    std::string content;
+    const KeyT& id() const {
+        return title;
+    }
+
+    friend husky::BinStream& operator<<(husky::BinStream& stream, const Document& doc) {
+        stream << doc.title << doc.content << doc.tf << doc.tf_idf << doc.words << doc.total_words;
+        return stream;
+    }
+    friend husky::BinStream& operator>>(husky::BinStream& stream, Document& doc) {
+        stream >> doc.title >> doc.content >> doc.tf >> doc.tf_idf >> doc.words >> doc.total_words;
+        return stream;
+    } 
+};
+
+class Term {
+  public:
+    using KeyT = std::string;
+    Term() = default;
+    explicit Term(const KeyT& term) : termid(term) {}
+    KeyT termid;
+    double idf;
+    const KeyT& id() const {
+        return termid;
+    }
+};
+
+
+void load_mongodb(const std::string& server, const std::string& db,
+    const std::string collection, const std::string user, const std::string pwd) {
+    auto & document_list = husky::ObjListFactory::create_objlist<Document>();
+    auto parse = [&] (std::string& chunk) {
+        mongo::BSONObj o = mongo::fromjson(chunk);
+        Document doc(o.getStringField("title"));
+        doc.content = o.getStringField("content");
+        document_list.add_object(doc);
+    };
+    husky::io::MongoDBInputFormat inputformat;
+    inputformat.set_server(server);
+    inputformat.set_ns(db, collection);
+    inputformat.set_auth(user, pwd);
+    load(inputformat, parse);
+    auto& term_list = husky::ObjListFactory::create_objlist<Term>();
+    husky::lib::Aggregator<int> totalDoc;
```

Comment:
Use `num_total_doc` (which means number of total documents) instead. We use `this_style_with_underline` for local variables.